fix: Update description for hobby upgrade script

danielthedifficult opened the pull request:
## Problem

In the  original hobby-upgrade warning's language (written by yours truly), I don't mention which version number nor which dates are relevant to the warning.

## Changes

[x] Added a check to see if the containers exist or not (`docker volume ls | grep -Pzoq 'root_cl2ickhouse-data\n(.|\n)*root_postgres-data\n'`) 
[x] Added the dates and version number now allows users to better understand if the warning is relevant to them. 
[x] Minor improvements to the language and instructions


## How did you test this code?

Tested in my own self-hosted PH hobby instance. 
The container checks work when containers are present, and fail when they are not.

danielthedifficult committed:
fix: Update description for hobby upgrade script

In the original warning's language (written by yours truly), I don't mention which version number nor which dates are relevant to the issue.

[x] Added a check to see if the containers exist or not (`docker volume ls | grep -Pzoq 'root_cl2ickhouse-data\n(.|\n)*root_postgres-data\n'`)
[x] Added the dates and version number now allows users to better understand if the warning is relevant to them.
[x] Minor improvements to the language and instructions

danielthedifficult commented:
Converted to draft, I forgot I'd need to update the second part of the warning as well.

danielthedifficult committed:
feat: track warning states for better user prompts

Use state from pre-update volume presence check to show post-install warning about data loss in addition to pre-upgrade warning.

danielthedifficult commented:
@hazzadous Given your history with this file, I'm pinging you to request a review and/or approval. Cheers!

hazzadous commented on bin/upgrade-hobby:
```diff
@@ -11,31 +11,41 @@ else
-read -r -p "Do you want to upgarde PostHog? [y/N] " response
-if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]
+echo "Checking for named postgres and clickhouse volumes to avoid data loss when upgrading from < 1.39"
+if docker volume ls | grep -Pzoq 'root_clickhouse-data\n(.|\n)*root_postgres-data\n'
```
hmm I have a feeling that the `root_` prepend is directory dependeny @danielthedifficult , am I wrong? I haven't checked. If not then all good.

danielthedifficult committed:
Improve references to docker container names

Previously instructions erroneously included my particular instance's containing folder. Changed grep check and added 'wildcards' in the instructions to clarify that it might be named differently depending on the environment.

fuziontech reviewed (approved):
Love it. Thanks @danielthedifficult !

fuziontech committed:
Merge branch 'master' into patch-1
